### PR TITLE
[RPRBLND-1704] Convert Image Texture to MaterialX

### PR DESCRIPTION
**Pull request by ardraeiss**
### PURPOSE
Convert Blender Image Texture shader node to MaterialX format.

### EFFECT OF CHANGE
- added Image Texture node support.

### TECHNICAL STEPS
- added Image Texture node converter;
- saved edited, packed and generated images to temporary files
- added Image Texture node to shader nodes menu.

### NOTES
- Tiled(UDIM) and Sequenced images aren't supported yet

### CODE REVIEW

File: src/hdusd/bl_nodes/nodes/texture.py [hunk 1]
```diff
@@ -12,12 +12,42 @@
...
 class ShaderNodeTexImage(NodeParser):
     def export(self):
-        return None
+        image_error_result = self.node_item((1.0, 0.0, 1.0, 1.0))
+        image = self.node.image
+        if not image:
+            log.warn(f"No image provided")
+            return image_error_result
+        # TODO support SEQUENCE image type
+        if image.source != 'FILE':
```
Comment (bnagirniak) [inline src/hdusd/bl_nodes/nodes/texture.py:29]: pixels can be exported to temporary image file like .exr. Look as example into export.material.sync_mx()

File: src/hdusd/bl_nodes/nodes/texture.py [hunk 2]
```diff
@@ -12,12 +12,42 @@
...
+            log.warn(f"Image {image} file doesn't exist at {file_path}")
+            return image_error_result
+        if image.size[0] * image.size[1] * image.channels == 0:
+            log.warn(f"Image {image} has no data")
+            return image_error_result
+
+        # TODO use Vector input for UV
+        uv = self.create_node('texcoord', 'vector2', {})
+
+        result = self.create_node('image', 'color4', {
```
Comment (bnagirniak) [inline src/hdusd/bl_nodes/nodes/texture.py:45]: color3?

**Review by bnagirniak — CHANGES_REQUESTED** [review]
commented in code

File: src/hdusd/bl_nodes/nodes/texture.py [hunk 3]
```diff
@@ -12,12 +12,42 @@
...
+import os
+
 from ..node_parser import NodeParser
+from . import log
 
 
 class ShaderNodeTexImage(NodeParser):
     def export(self):
-        return None
+        image_error_result = self.node_item((1.0, 0.0, 1.0, 1.0))
```
[the remark this answers is not in the document]
Comment (ardraeiss) [reply src/hdusd/bl_nodes/nodes/texture.py]: Extracted to other image textures-related constants in the same module

**Review by bnagirniak — APPROVED** [review]
Code is ok.

**Pull request merged**